core/src/main/java/org/apache/accumulo/core/metadata/schema/SelectedFiles.java

```diff
@@ -151,7 +133,29 @@ public long getFateTxId() {
   }
 
   public String getMetadataValue() {
-    return metadataValue;
+    if (this.metadataValue == null) {
+      // use the custom TypeAdapter to create the json
+      this.metadataValue = GSON.toJson(this);
+    }
+    return this.metadataValue;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    SelectedFiles other = (SelectedFiles) obj;
+    return fateTxId == other.fateTxId && Objects.equals(files, other.files)
```

Comment:
Seems like the following would be safe because it never expected that files would null.

```suggestion
    return fateTxId == other.fateTxId && files.equals(other.files)
```